src/Database/LSMTree/Internal/Config.hs

```diff
@@ -95,6 +97,16 @@ defaultTableConfig =
       , confMergeSchedule     = defaultMergeSchedule
       }
 
+data RunLevelNo = RegularLevel LevelNo | UnionLevel
+
+runParamsForLevel :: TableConfig -> RunLevelNo -> RunParams
+runParamsForLevel conf@TableConfig {..} levelNo =
+    RunParams
+      { runParamCaching = diskCachePolicyForLevel confDiskCachePolicy levelNo
+      , runParamAlloc   = bloomFilterAllocForLevel conf levelNo
+      , runParamIndex   = indexTypeForRun confFencePointerIndex
+      }
```

Comment:
I define this in one place for both regular and union levels. Does that seem sensible?